Mathlib/Analysis/Distribution/ContDiffMapSupportedIn.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2023 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker, Luigi Massacci
+-/
+
+import Mathlib.Analysis.Calculus.ContDiff.Defs
+import Mathlib.Topology.ContinuousMap.Bounded.Normed
+import Mathlib.Topology.Sets.Compacts
+
+/-!
+# Continuously differentiable functions supported in a given compact
+
+This file develops the basic theory of bundled `n`-times continuously differentiable functions
+with support contained in a given compact.
+
+Given `n : ℕ∞` and a compact subset `K` of a normed space `E`, we consider the type of bundled
+functions `f : E → F` (where `F` is a normed vector space) such that:
+
+- `f` is `n`-times continuously differentiable: `ContDiff ℝ n f`.
+- `f` vanishes outside of a compact: `EqOn f 0 Kᶜ`.
+
+We define it as a bundled structure so as to endow it with the topology required to make it a ù
+subspace of the space of test functions.
+
+## Main definitions
```

Comment:
I would add to this section (maybe in a future PR) the definition of the topology, with an explanation of what it is ("uniform convergence of `f` and its derivatives up to order `n`).